types/schemas/app/index.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ApplicationTypeKeys,
+  PrimaryApplicationType,
+} from '../application/enums/ApplicationTypes';
+import {ApplicationData} from './ApplicationData';
+import {User} from './User';
+
+/**
+ * @internal
+ * The generic base type for all applications
+ * Takes a primary and granular application type which allows child properties to differ based on these inputs
+ * Deriving `TPrimary` from `TGranular` is possible in TS, but not in a way which is currently compatible with ts-json-schema-generator
+ */
+interface ApplicationSpecification<
+  TPrimary extends PrimaryApplicationType,
+  TGranular extends ApplicationTypeKeys,
+> {
+  applicationType: TGranular;
+  data: {
+    user: User<TPrimary>;
+    application: ApplicationData<TPrimary>;
+  };
+}
+
+export type PlanningPermissionApplication = ApplicationSpecification<
+  'pp',
+  'pp'
+>;
+// TODO: All granular types
+
+export type PriorApprovalApplication = ApplicationSpecification<'pa', 'pa'>;
+// TODO: All granular types
+
+export type WorksToTreesApplications = ApplicationSpecification<
+  'wtt',
+  Extract<ApplicationTypeKeys, 'wtt' | 'wtt.consent' | 'wtt.notice'>
+>;
+
+/**
+ * @title App
+ * @description
+ * (Temporary name to not clash with the existing `Application` type)
+ * The root specification for a planning application in England generated by a digital planning service
+ */
+export type App =
```

Comment:
nit: can we call this "NextApp" or "TestApp" or "PrototypeApp" or something similar that makes it clear it's work-in-progress adn not just short-hand for existing "Application" ?

Reply:
Very good shout - will make this change before I merge.